### PR TITLE
Sorting blindedMessages by amount

### PR DESCRIPTION
**Pull request by Egge21M**
# Fixes: #90

## Description
Ordered blindedMessages returned by createSplitPayload by amount and added a keepVector

## Changes

- Ordered blindedMessages by amount
- added keepVector as return value

## PR Tasks

- [x] Open PR
- [x] run `npm run test` --> no failing unit tests
- [x] run `npm run format`

### CODE REVIEW

**Comment by Egge21M** [comment]
This is now ready for review. As outline above I changed createSplitPayload to return a payload containing a list of proofs ordered by amount, as well as a list of booleans indicating which proof to keep and which to send.
I then changed send to use the list of booleans (keepVector) to /split and keep the right proofs.

@gandlafbtc @callebtc

File: src/CashuWallet.ts [hunk 1]
```diff
@@ -474,22 +473,43 @@ class CashuWallet {
...
+
+		const convertedSendMessages = sendBlindedMessages.blindedMessages.map((_, i) => ({
+			msg: sendBlindedMessages.blindedMessages[i],
+			secret: sendBlindedMessages.secrets[i],
+			rs: sendBlindedMessages.rs[i],
+			amount: sendBlindedMessages.amounts[i],
+			keep: false
+		}));
+		const keepVector: Array<boolean> = [];
 		const blindedMessages: BlindedTransaction = {
```
Comment (gandlafbtc) [inline src/CashuWallet.ts:492]: let's rename this variable to blindedTransaction. `blindedMessages` seems kinda confusing. What do you think?

File: src/CashuWallet.ts [hunk 2]
```diff
@@ -474,22 +473,43 @@ class CashuWallet {
...
-				...keepBlindedMessages.blindedMessages,
-				...sendBlindedMessages.blindedMessages
-			],
-			secrets: [...keepBlindedMessages.secrets, ...sendBlindedMessages.secrets],
-			rs: [...keepBlindedMessages.rs, ...sendBlindedMessages.rs],
-			amounts: [...keepBlindedMessages.amounts, ...sendBlindedMessages.amounts]
+			blindedMessages: [],
+			secrets: [],
+			rs: [],
+			amounts: []
```
Comment (gandlafbtc) [inline src/CashuWallet.ts:496]: we could add the keep vector here instead

**Review by gandlafbtc — COMMENTED** [review]
Looks good! Only thing missing is a Unit test testing the behaviour.

File: src/CashuWallet.ts (hunk 1 above)
Comment (Egge21M) [reply src/CashuWallet.ts:492]: Oh yes absolutely. Will do

File: src/CashuWallet.ts (hunk 2 above)
Comment (Egge21M) [reply src/CashuWallet.ts:496]: True, I was not sure how many lines of code depend on the type  `BlindedTransaction` and wanted to reduce this PRs footprint. But I will check that out in my next iteration.

**Comment by Egge21M** [comment]
> Looks good! Only thing missing is a Unit test testing the behaviour.

I was under the assumption that the old units tests for this method would still suffice, but we should probably add one to check wether the amount is actually sorted (will do asap). Or am I missing something?

**Comment by Egge21M** [comment]
> Looks good! Only thing missing is a Unit test testing the behaviour.

What would be good way to easily generate secrets and signatures for the Unit tests?

**Comment by gandlafbtc** [comment]
@Egge21M I think this PR might be outdated, especially after https://github.com/cashubtc/cashu-ts/pull/159.. shall we close it?

**Pull request closed without merging**